src/components/Button/Button.tsx

```diff
@@ -119,22 +165,37 @@ const ButtonComponent: FC<PropsWithChildrenAndClass<Props>> = ({
       },
     ],
     defaultVariants: {
-      intent: 'primary',
+      intent: buttonGroup.intent ?? 'primary',
       isSquare: false,
       hasSmallPaddings: false,
-      size: 'md',
+      size: buttonGroup.size ?? 'md',
     },
   });
 
+  if (isLink && href != null) {
+    return (
+      <Link
+        href={href}
+        external={external}
+        isUnstyled
+        className={cx(button(props))}
+      >
+        {leftElement && leftElement}
+        {children}
+      </Link>
+    );
+  }
```

Comment:
I think `isLink` prop can be replaced with `as`/`Component` prop. By default it will be button, but we can pass `Link`. Mui `ListItem` works the same way: we can pass any link component and it will automatically recognize it's props. You can take `Field` component as an example of polymorphic component. 

But if you think it's an overengineering, then let's leave it as-is